processor/src/host/mod.rs

```diff
@@ -32,6 +33,8 @@ pub trait BaseHost {
     // REQUIRED METHODS
     // --------------------------------------------------------------------------------------------
 
+    fn source_manager(&self) -> Arc<dyn SourceManagerSync>;
```

Comment:
```suggestion
    fn source_manager(&self) -> &Arc<dyn SourceManagerSync>;
```
nit: avoids cloning unnecessarily if not needed, and allows to e.g. implement a method

```rs
fn manager_source_slice(&self) -> &str {
    // if `source_manager()` returns `Arc<T>` instead of `&Arc<T>`, then this fails to compile
    // because we return a reference to a temporary value
    self.source_manager().source_slice().unwrap()
}
```

Reply:
Any reason we need to be returning a reference to an `Arc` here? Why not just return `&dyn SourceManagerSync`? I think it would be best for us to discourage threading `Arc`'s through things like the assembler and processor, and it saves us a layer of indirection, as a reference to an Arc would have us deref once to get the Arc, then deref again to get the trait object, then invoke whatever trait method via the function pointer from the trait object. That's a lot of pointer chasing for the off-chance we want to clone the `Arc`